src/lib/storage/table.hpp

```diff
@@ -243,6 +254,11 @@ class Table : private Noncopyable {
   tbb::concurrent_vector<std::shared_ptr<Chunk>, ZeroAllocator<std::shared_ptr<Chunk>>> _chunks;
 
   TableKeyConstraints _table_key_constraints;
+  TableOrderConstraints _table_order_constraints;
+  ForeignKeyConstraints _foreign_key_constraints;
+  // Stores the ForeignKeyCostraints of another table that reference this table. Required for maintenance of inclusion
```

Comment:
Ahh ... now I somewhat understand why the code above talked about INDs. Can you elaborate a bit about it? Otherwise it's pretty confusing.

Reply:
Might also add something to `AbstractTableConstraint` that our constraints are structures on table level that are used to represent / are translated to data dependencies on LQP level.

Reply:
Might, will, or have? :)

Reply:
As for now, I already have:

https://github.com/hyrise/hyrise/blob/9235278e8b85861c78e11a7337a1abbdf87f1603/src/lib/storage/constraints/abstract_table_constraint.hpp#L8-L15